### DISCUSSION TITLE
Make zod support official, since the original package is no longer maintained.

### DISCUSSION THREAD

**Discussion by Bram-dc** [fastify/fastify#5709] [General] [answered]
The original type provider (https://github.com/turkerdev/fastify-type-provider-zod) is not being maintained at the moment. So users are unable to use it with the newly released fastify v5. Nearly 3 months ago I created the PR to support this feature: https://github.com/turkerdev/fastify-type-provider-zod/pull/97.

Recently, I have continued working on improving the type-provider and created my own release: [Github](https://github.com/bram-dc/fastify-type-provider-zod) & [NPM](https://www.npmjs.com/package/@bram-dc/fastify-type-provider-zod). This release has some improvements over the old package such as using the internal fastify error system, correctly throwing validation errors and support for refs.

I was wondering if we could maybe instead release this package under the official @fastify/type-provider-zod instead. What would be the steps involved into doing this?

**Comment by Uzlopak** [comment]
@kibertoad 

As the maintainer of said package, do you concur that our zod package is unmaintained?

**Comment by kibertoad** [comment]
No, I'm just pretty busy 😅 
That said, I'll release the fastify v5 support today.

**Reply by Bram-dc** [reply]
Great, thanks!

**Reply by Uzlopak** [reply]
Also you are invited to provide PRs to our core plugin or atleast let us discuss your changes in issues

**Comment by kibertoad** [comment, marked as answer]
3.0.0 was released with fastify 5 support

**Comment by Xhale1** [comment]
I believe there is still value in publishing the zod type provider package under the `@fastify/` npm scope. Is that still a possibility?

Given both TypeBox and Zod are mentioned in Fastify's docs, I think it's reasonable to have both type providers appear equally official.

**Reply by Bram-dc** [reply]
I think that is also what we should do.
